packages/cozy-client/src/models/folder.js

```diff
@@ -0,0 +1,111 @@
+const APP_DOCTYPE = 'io.cozy.apps'
+
+export const MAGIC_FOLDERS = {
+  ADMINISTRATIVE: `${APP_DOCTYPE}/administrative`,
+  PHOTOS: `${APP_DOCTYPE}/photos`,
+  PHOTOS_BACKUP: `${APP_DOCTYPE}/photos/mobile`,
+  PHOTOS_UPLOAD: `${APP_DOCTYPE}/photos/upload`,
+  NOTES_FOLDER: `${APP_DOCTYPE}/notes`,
+  HOME_FOLDER: `${APP_DOCTYPE}/home`
+}
+
+/**
+ * Create a folder with a reference to the given document
+ *
+ * @param  {object}  client   cozy-client instance
+ * @param  {string}  path     Folder path
+ * @param  {object}  document Document to make reference to. Any doctype.
+ * @returns {object}  Folder document
+ */
+export const createFolderWithReference = async (client, path, document) => {
+  const collection = client.collection('io.cozy.files')
+  const dirId = await collection.ensureDirectoryExists(path)
+  await collection.addReferencesTo(document, [
+    {
+      _id: dirId
+    }
+  ])
+
+  const { data: dirInfos } = await collection.get(dirId)
+
+  return dirInfos
+}
+
+/**
+ * Returns a "Magic Folder", given its id
```

Comment:
can we add a @see https://docs.cozy.io/en/cozy-doctypes/docs/io.cozy.apps/#special-iocozyapps-doctypes ?